app/models/bike.rb

```diff
@@ -497,6 +497,27 @@ def external_image_urls
     b_params.map { |bp| bp.external_image_urls }.flatten.reject(&:blank?).uniq
   end
 
+  def alert_image_url
+    return if current_stolen_record.blank?
+
+    generate_alert_image
+    current_stolen_record.alert_image_url
+  end
+
+  def generate_alert_image
+    return if current_stolen_record.blank?
+    return if current_stolen_record.alert_image.present?
+
+    # Remove any old alert images
+    stolen_records.each { |sr| sr.alert_image&.remove! }
+
+    return if public_images.none?
+
+    # Generate alert image for current stolen record
+    current_stolen_record.alert_image = public_images.first.image
```

Comment:
Why is this defined here, but also in the uploader?

Reply:
Assigning the image to `alert_image` and saving here triggers the uploader to run.
Having some trouble parsing the question though, let me know if I'm missing something.

Reply:
This is mostly a style nit, not a serious concern.

`def bike_image` in `AlertImageUploader` actually chooses what image is processed. But because this assigns to `stolen_record.alert_image`, I would assume this method in `bike.rb` is what chooses the image to process. 

This is a response to this imaginary scenario: I come back to this code and want to update which image is processed - so I change it here, and am confused about why it doesn't change which image ends up being processed.

This is a nit you can ignore, since the above isn't actually going to happen.

Reply:
Ah thanks for clarifying, that's a good call. `AlertImageUploader#bike_image` is vestigial, I'll get rid of it 👍